## Project Objectives:
1. Find the percentage of all licensed sets ever Star Wars themed licensed released.

2. Find in which year was not the Star Wars licensed theme the most popular (in terms of number of sets released that particular year).

3. Determine how many unique datasets were released each year (1955-2017).

In [ ]:
import pandas as pd

df = pd.read_csv('https://raw.githubusercontent.com/KeithGalli/lego-analysis/master/datasets/lego_sets.csv')
parent_theme = pd.read_csv('https://raw.githubusercontent.com/KeithGalli/lego-analysis/master/datasets/parent_themes.csv')

In [ ]:
df.head( )

set_num                        name  year  num_parts    theme_name  \
0    00-1             Weetabix Castle  1970      471.0        Castle   
1  0011-2           Town Mini-Figures  1978        NaN  Supplemental   
2  0011-3  Castle 2 for 1 Bonus Offer  1987        NaN  Lion Knights   
3  0012-1          Space Mini-Figures  1979       12.0  Supplemental   
4  0013-1          Space Mini-Figures  1979       12.0  Supplemental   

  parent_theme  
0     Legoland  
1         Town  
2       Castle  
3        Space  
4        Space

In [ ]:
parent_theme.head(50 )

id                      name  is_licensed
0     1                   Technic        False
1    22                   Creator        False
2    50                      Town        False
3   112                    Racers        False
4   126                     Space        False
5   147                   Pirates        False
6   155         Modular Buildings        False
7   158                 Star Wars         True
8   186                    Castle        False
9   204             Designer Sets        False
10  206                  Seasonal        False
11  233                     Train        False
12  246              Harry Potter         True
13  252              Architecture        False
14  254               Bulk Bricks        False
15  258                Mindstorms        False
16  263  Pirates of the Caribbean         True
17  264             Indiana Jones         True
18  269                      Cars         True
19  270                    Ben 10         True
20  271          Prince of Persia         True
21  272     SpongeBob SquarePants         True
22  273                   Studios        False
23  275                 Toy Story         True
24  276                Sculptures        False
25  278                Model Team        False
26  279                 4 Juniors        False
27  296               Adventurers        False
28  301                     Other        False
29  302                    Agents        False
30  304                Alpha Team        False
31  307                  Aquazone        False
32  315                  Atlantis        False
33  317                    Avatar         True
34  318                  Belville        False
35  324                  Bionicle        False
36  363                      Boat        False
37  364  Building Set with People        False
38  365                   Classic        False
39  384                 Dino 2010        False
40  385               Dino Attack        False
41  386                 Dinosaurs        False
42  387                 Discovery        False
43  388     Disney's Mickey Mouse         True
44  389                 Exo-Force        False
45  390                  Fabuland        False
46  397                   Factory        False
47  398         FIRST LEGO League        False
48  399                 Freestyle        False
49  400              Hero Factory        False

In [ ]:
merged = df.merge(parent_theme, left_on='parent_theme', right_on="name")
merged.drop(columns= 'name_y', inplace=True)
merged.head()

set_num                             name_x  year  num_parts theme_name  \
0    00-1                    Weetabix Castle  1970      471.0     Castle   
1    00-2       Weetabix Promotional House 1  1976        NaN   Building   
2    00-3       Weetabix Promotional House 2  1976        NaN   Building   
3    00-4      Weetabix Promotional Windmill  1976      126.0   Building   
4    00-7  Weetabix Promotional Lego Village  1976        NaN   Building   

  parent_theme   id  is_licensed  
0     Legoland  411        False  
1     Legoland  411        False  
2     Legoland  411        False  
3     Legoland  411        False  
4     Legoland  411        False

In [ ]:
licensed = merged[merged['is_licensed']==True]
licensed.head()

star_wars = licensed[licensed['parent_theme']=='Star Wars']

type(star_wars.shape[0])



int

In [ ]:
licensed.shape[0]

1332

In [ ]:
star_wars = licensed[licensed['parent_theme']=='Star Wars']
star_wars

set_num                                   name_x  year  num_parts  \
3493  10018-1                               Darth Maul  2001     1868.0   
3494  10019-1              Rebel Blockade Runner - UCS  2001        NaN   
3495  10026-1                  Naboo Starfighter - UCS  2002        NaN   
3496  10030-1            Imperial Star Destroyer - UCS  2002     3115.0   
3497  10123-1                               Cloud City  2003      707.0   
...       ...                                      ...   ...        ...   
4097    VP-12       Star Wars Co-Pack of 7121 and 7151  2000        2.0   
4098     VP-2       Star Wars Co-Pack of 7110 and 7144  2001        2.0   
4099     VP-3       Star Wars Co-Pack of 7131 and 7151  2000        2.0   
4100     VP-4  Star Wars Co-Pack of 7101 7111 and 7171  2000        3.0   
4101     VP-8       Star Wars Co-Pack of 7130 and 7150  2000        NaN   

                   theme_name parent_theme   id  is_licensed  
3493                Star Wars    Star Wars  158         True  
3494  Star Wars Episode 4/5/6    Star Wars  158         True  
3495      Star Wars Episode 1    Star Wars  158         True  
3496  Star Wars Episode 4/5/6    Star Wars  158         True  
3497  Star Wars Episode 4/5/6    Star Wars  158         True  
...                       ...          ...  ...          ...  
4097      Star Wars Episode 1    Star Wars  158         True  
4098  Star Wars Episode 4/5/6    Star Wars  158         True  
4099      Star Wars Episode 1    Star Wars  158         True  
4100      Star Wars Episode 1    Star Wars  158         True  
4101  Star Wars Episode 4/5/6    Star Wars  158         True  

[609 rows x 8 columns]

In [ ]:
the_force = star_wars.shape[0]/licensed.shape[0]
print(the_force)  #Comprobamos que se dejen operaar

0.4572072072072072


In [ ]:
merged[merged['set_num'].isnull()].shape

(153, 8)

In [ ]:
licensed = licensed.dropna(subset=['set_num'])
licensed

set_num                         name_x  year  num_parts  \
3493   10018-1                     Darth Maul  2001     1868.0   
3494   10019-1    Rebel Blockade Runner - UCS  2001        NaN   
3495   10026-1        Naboo Starfighter - UCS  2002        NaN   
3496   10030-1  Imperial Star Destroyer - UCS  2002     3115.0   
3497   10123-1                     Cloud City  2003      707.0   
...        ...                            ...   ...        ...   
11981   8410-1                      Swampfire  2010       22.0   
11982   8411-1                    ChromaStone  2010       21.0   
11983   8517-1                   Humungousaur  2010       14.0   
11984   8518-1                        Jet Ray  2010        NaN   
11985   8519-1                      Big Chill  2010       20.0   

                    theme_name parent_theme   id  is_licensed  
3493                 Star Wars    Star Wars  158         True  
3494   Star Wars Episode 4/5/6    Star Wars  158         True  
3495       Star Wars Episode 1    Star Wars  158         True  
3496   Star Wars Episode 4/5/6    Star Wars  158         True  
3497   Star Wars Episode 4/5/6    Star Wars  158         True  
...                        ...          ...  ...          ...  
11981                   Ben 10       Ben 10  270         True  
11982                   Ben 10       Ben 10  270         True  
11983                   Ben 10       Ben 10  270         True  
11984                   Ben 10       Ben 10  270         True  
11985                   Ben 10       Ben 10  270         True  

[1179 rows x 8 columns]

In [ ]:
the_force = star_wars.shape[0]/licensed.shape[0]
print(the_force)  #Comprobamos que se dejen operaar

0.5165394402035624


In [ ]:
the_force = int((star_wars.shape[0]/licensed.shape[0])*100)
print(the_force)  #Encontramos el valor

51


In [ ]:
the_force = ((star_wars.shape[0]/licensed.shape[0])*100)
print(the_force)

51.653944020356235


## First answer
1) The percentage of Star Wars Licensed sales over all Licensed products is 51.65%, that is, the majority of Licensed Legos are from this franchise.

## Solving the second task




In [ ]:
licensed_sorted = licensed.sort_values('year')
#Acá simplemente estamos explorando
licensed_sorted.groupby(['year','parent_theme']).head(50)

set_num                       name_x  year  num_parts  \
3702   7161-1                   Gungan Sub  1999      379.0   
3705   7171-1             Mos Espa Podrace  1999        NaN   
3690   7140-1               X-wing Fighter  1999      271.0   
3685   7130-1                  Snowspeeder  1999        NaN   
3684   7128-1                Speeder Bikes  1999       93.0   
...       ...                          ...   ...        ...   
5678  70912-1                Arkham Asylum  2017     1621.0   
5679  70913-1  Scarecrow" Fearful Face-off  2017        NaN   
5680  70914-1     Bane" Toxic Truck Attack  2017      365.0   
5681  70915-1  Two-Face" Double Demolition  2017      563.0   
5682  70916-1                  The Batwing  2017     1052.0   

                   theme_name  parent_theme   id  is_licensed  
3702      Star Wars Episode 1     Star Wars  158         True  
3705      Star Wars Episode 1     Star Wars  158         True  
3690  Star Wars Episode 4/5/6     Star Wars  158         True  
3685  Star Wars Episode 4/5/6     Star Wars  158         True  
3684  Star Wars Episode 4/5/6     Star Wars  158         True  
...                       ...           ...  ...          ...  
5678                   Batman  Super Heroes  482         True  
5679                   Batman  Super Heroes  482         True  
5680                   Batman  Super Heroes  482         True  
5681                   Batman  Super Heroes  482         True  
5682                   Batman  Super Heroes  482         True  

[1133 rows x 8 columns]

In [ ]:
licensed_sorted = licensed.sort_values('year')
licensed_sorted['count'] = 1

summed_df = licensed_sorted.groupby(['year','parent_theme']).sum().reset_index()

summed_df.head()

year           parent_theme  \
0  1999              Star Wars   
1  2000  Disney's Mickey Mouse   
2  2000              Star Wars   
3  2001           Harry Potter   
4  2001              Star Wars   

                                             set_num  \
0  7161-17171-17140-17130-17128-17141-17121-17131...   
1                     4165-14167-14166-14164-14178-1   
2  7134-13341-17159-17144-17124-17115-17104-14151...   
3  4723-14707-14722-14721-14709-14701-14702-14704...   
4  7166-17186-165028-165030-165034-18007-18008-17...   

                                              name_x  num_parts  \
0  Gungan SubMos Espa PodraceX-wing FighterSnowsp...     1384.0   
1  Minnie's Birthday PartyMickey's MansionMickey'...      405.0   
2  A-wing FighterStar Wars #2 - Luke/Han/Boba Min...     2580.0   
3  Diagon Alley ShopsHagrid's HutGryffindorHogwar...     1284.0   
4  Imperial ShuttleWatto's JunkyardStar Wars Co-P...     2949.0   

                                          theme_name    id  is_licensed  count  
0  Star Wars Episode 1Star Wars Episode 1Star War...  2054           13     13  
1  Disney's Mickey MouseDisney's Mickey MouseDisn...  1940            5      5  
2  Star Wars Episode 4/5/6Star Wars Episode 4/5/6...  4108           26     26  
3  Sorcerer's StoneSorcerer's StoneSorcerer's Sto...  2706           11     11  
4  Star Wars Episode 4/5/6Star Wars Episode 1Star...  2212           14     14

In [ ]:
summed_dfp = licensed_sorted.groupby(['year','parent_theme']).sum()
summed_dfp

set_num  \
year parent_theme                                                                  
1999 Star Wars                 7161-17171-17140-17130-17128-17141-17121-17131...   
2000 Disney's Mickey Mouse                        4165-14167-14166-14164-14178-1   
     Star Wars                 7134-13341-17159-17144-17124-17115-17104-14151...   
2001 Harry Potter              4723-14707-14722-14721-14709-14701-14702-14704...   
     Star Wars                 7166-17186-165028-165030-165034-18007-18008-17...   
...                                                                          ...   
2017 Disney Princess                  41143-141148-141147-141146-141145-141144-1   
     Minecraft                 21137-121135-121136-121134-121133-121132-12113...   
     Pirates of the Caribbean                                            71042-1   
     Star Wars                 75531-175162-175161-175160-17516075144-1911618...   
     Super Heroes              76075-1211701-130449-130521-17090470903-170902...   

                                                                          name_x  \
year parent_theme                                                                  
1999 Star Wars                 Gungan SubMos Espa PodraceX-wing FighterSnowsp...   
2000 Disney's Mickey Mouse     Minnie's Birthday PartyMickey's MansionMickey'...   
     Star Wars                 A-wing FighterStar Wars #2 - Luke/Han/Boba Min...   
2001 Harry Potter              Diagon Alley ShopsHagrid's HutGryffindorHogwar...   
     Star Wars                 Imperial ShuttleWatto's JunkyardStar Wars Co-P...   
...                                                                          ...   
2017 Disney Princess           Berry's KitchenElsa's Magical Ice PalaceAnna's...   
     Minecraft                 The Mountain CaveThe Crafting Box 2.0The Ocean...   
     Pirates of the Caribbean                                        Silent Mary   
     Star Wars                 Stormtrooper" CommanderY-wingTie Striker Micro...   
     Super Heroes              Wonder Woman" Warrior BattleBatmanThe MilanoTh...   

                               num_parts  \
year parent_theme                          
1999 Star Wars                    1384.0   
2000 Disney's Mickey Mouse         405.0   
     Star Wars                    2580.0   
2001 Harry Potter                 1284.0   
     Star Wars                    2949.0   
...                                  ...   
2017 Disney Princess               347.0   
     Minecraft                    5264.0   
     Pirates of the Caribbean     2286.0   
     Star Wars                    7583.0   
     Super Heroes                13123.0   

                                                                      theme_name  \
year parent_theme                                                                  
1999 Star Wars                 Star Wars Episode 1Star Wars Episode 1Star War...   
2000 Disney's Mickey Mouse     Disney's Mickey MouseDisney's Mickey MouseDisn...   
     Star Wars                 Star Wars Episode 4/5/6Star Wars Episode 4/5/6...   
2001 Harry Potter              Sorcerer's StoneSorcerer's StoneSorcerer's Sto...   
     Star Wars                 Star Wars Episode 4/5/6Star Wars Episode 1Star...   
...                                                                          ...   
2017 Disney Princess           Disney PrincessDisney PrincessDisney PrincessD...   
     Minecraft                 MinecraftMinecraftMinecraftMinecraftMinecraftM...   
     Pirates of the Caribbean                           Pirates of the Caribbean   
     Star Wars                 Star WarsStar WarsMiniStar Wars Rogue OneStar ...   
     Super Heroes              Super HeroesBatmanMarvelBatmanBatmanBatmanBatm...   

                                  id  is_licensed  count  
year parent_theme                                         
1999 Star Wars                  2054           13     13  
2000 Disney's Mickey Mouse      1940        

In [ ]:
max_df = summed_df.sort_values('count', ascending = False).drop_duplicates(['year'])

max_df

year  parent_theme                                            set_num  \
82  2017  Super Heroes  76075-1211701-130449-130521-17090470903-170902...   
76  2016     Star Wars  75156-175157-175159-130279-175158-130602-13060...   
67  2015     Star Wars  75099-1SW911506-1SW911508-1SW911509-1SW911511-...   
59  2014     Star Wars  75046-175045-175044-175043-175039-175041-17504...   
47  2012     Star Wars  66432-1comcon019-1comcon024-130058-166431-1949...   
32  2009     Star Wars  8039-18033-17748-17749-110198-17751-17752-1775...   
52  2013     Star Wars  75001-175000-166449-130240-130241-130242-13024...   
9   2003     Star Wars  K4487-1K4488-1K4482-1VP-10K4479-1K4480-14487-1...   
42  2011     Star Wars  66396-166395-166378-166377-1comcon015-17915-17...   
36  2010     Star Wars  8099-18084-18085-18086-18087-18088-18089-18091...   
16  2005     Star Wars  7259-17258-17256-17255-17252-17257-16967-27261...   
6   2002     Star Wars  8011-165153-165145-165106-1K7204-1K7153-17163-...   
2   2000     Star Wars  7134-13341-17159-17144-17124-17115-17104-14151...   
28  2008     Star Wars  comcon001-17667-17668-17669-17669-27670-17670-...   
12  2004     Star Wars  10134-17262-16964-26963-210131-1K4492-16963-1K...   
24  2007     Star Wars  7660-17659-17658-17657-17656-17661-17662-17663...   
4   2001     Star Wars  7166-17186-165028-165030-165034-18007-18008-17...   
0   1999     Star Wars  7161-17171-17140-17130-17128-17141-17121-17131...   
20  2006     Star Wars  10175-110174-16205-16208-16206-16211-16212-172...   

                                               name_x  num_parts  \
82  Wonder Woman" Warrior BattleBatmanThe MilanoTh...    13123.0   
76  Krennic’s Imperial ShuttleCaptain Rex’s AT-TED...     6934.0   
67  Rey's Speeder™SnowspeederMini Slave 1Imperial ...    11410.0   
59  Coruscant Police GunshipRepublic AV-7 Anti-Veh...     8293.0   
47  Star Wars Super Pack 3 in 1 (9490, 9492, 9496)...     6769.0   
32  Venator-Class Republic Attack CruiserGeneral G...     3953.0   
52  Republic Troopers vs. Sith TroopersClone Troop...     6159.0   
9   Star Wars Miniatures Kit IStar Wars Miniatures...     6660.0   
42  Star Wars Super Pack 3 in 1 (7877 7929 7913)St...     3451.0   
36  Midi-Scale Imperial Star DestroyerSnowtrooper ...     6003.0   
16  ARC-170 StarfighterWookiee AttackJedi Starfigh...     4730.0   
6   Jango Fett™Jango Fett's Slave I (Set 7153) wit...     4735.0   
2   A-wing FighterStar Wars #2 - Luke/Han/Boba Min...     2580.0   
28  Clone Wars Pack - San Diego Comic-Con 2008 Exc...     6865.0   
12  Y-wing Attack Starfighter - UCSTIE Fighter and...     1659.0   
24  Naboo N-1 Starfighter and Vulture DroidImperia...    11361.0   
4   Imperial ShuttleWatto's JunkyardStar Wars Co-P...     2949.0   
0   Gungan SubMos Espa PodraceX-wing FighterSnowsp...     1384.0   
20  Vader's TIE Advanced - UCSImperial AT-ST - UCS...     2769.0   

                                           theme_name     id  is_licensed  \
82  Super HeroesBatmanMarvelBatmanBatmanBatmanBatm...  34704           72   
76  Star Wars Rogue OneStar Wars RebelsStar WarsSt...   9638           61   
67  Star WarsStar Wars Episode 4/5/6MiniStar Wars ...   9164           58   
59  Star WarsStar WarsStar WarsStar WarsStar WarsS...   7110           45   
47  Star WarsStar WarsStar Wars Episode 4/5/6Star ...   6794           43   
32  Star Wars Clone WarsStar Wars Episode 3Star Wa...   6162           39   
52  Star WarsStar Wars Episode 2Star WarsStar Wars...   5530           35   
9   MiniMiniStar Wars Episode 2Star Wars OtherStar...   5056           32   
42  Star Wars Episode 1Star WarsStar Wars Clone Wa...   5056           32   
36  Star Wars Episode 4/5/6Star Wars Episode 4/5/6...   4740           30   
16  Star Wars Episode 3Star Wars Episode 3Star War...   4424           28   
6   Star WarsStar Wars Episode 2Star Wars Episode ...   4424           28   
2   Star Wars Episode 4/5/6Star Wars Episode 4/5/6...   4108           26   
28  Star Wars Clone Wars

In [ ]:
max_df.sort_values('year', inplace = True)
max_df

year  parent_theme                                            set_num  \
0   1999     Star Wars  7161-17171-17140-17130-17128-17141-17121-17131...   
2   2000     Star Wars  7134-13341-17159-17144-17124-17115-17104-14151...   
4   2001     Star Wars  7166-17186-165028-165030-165034-18007-18008-17...   
6   2002     Star Wars  8011-165153-165145-165106-1K7204-1K7153-17163-...   
9   2003     Star Wars  K4487-1K4488-1K4482-1VP-10K4479-1K4480-14487-1...   
12  2004     Star Wars  10134-17262-16964-26963-210131-1K4492-16963-1K...   
16  2005     Star Wars  7259-17258-17256-17255-17252-17257-16967-27261...   
20  2006     Star Wars  10175-110174-16205-16208-16206-16211-16212-172...   
24  2007     Star Wars  7660-17659-17658-17657-17656-17661-17662-17663...   
28  2008     Star Wars  comcon001-17667-17668-17669-17669-27670-17670-...   
32  2009     Star Wars  8039-18033-17748-17749-110198-17751-17752-1775...   
36  2010     Star Wars  8099-18084-18085-18086-18087-18088-18089-18091...   
42  2011     Star Wars  66396-166395-166378-166377-1comcon015-17915-17...   
47  2012     Star Wars  66432-1comcon019-1comcon024-130058-166431-1949...   
52  2013     Star Wars  75001-175000-166449-130240-130241-130242-13024...   
59  2014     Star Wars  75046-175045-175044-175043-175039-175041-17504...   
67  2015     Star Wars  75099-1SW911506-1SW911508-1SW911509-1SW911511-...   
76  2016     Star Wars  75156-175157-175159-130279-175158-130602-13060...   
82  2017  Super Heroes  76075-1211701-130449-130521-17090470903-170902...   

                                               name_x  num_parts  \
0   Gungan SubMos Espa PodraceX-wing FighterSnowsp...     1384.0   
2   A-wing FighterStar Wars #2 - Luke/Han/Boba Min...     2580.0   
4   Imperial ShuttleWatto's JunkyardStar Wars Co-P...     2949.0   
6   Jango Fett™Jango Fett's Slave I (Set 7153) wit...     4735.0   
9   Star Wars Miniatures Kit IStar Wars Miniatures...     6660.0   
12  Y-wing Attack Starfighter - UCSTIE Fighter and...     1659.0   
16  ARC-170 StarfighterWookiee AttackJedi Starfigh...     4730.0   
20  Vader's TIE Advanced - UCSImperial AT-ST - UCS...     2769.0   
24  Naboo N-1 Starfighter and Vulture DroidImperia...    11361.0   
28  Clone Wars Pack - San Diego Comic-Con 2008 Exc...     6865.0   
32  Venator-Class Republic Attack CruiserGeneral G...     3953.0   
36  Midi-Scale Imperial Star DestroyerSnowtrooper ...     6003.0   
42  Star Wars Super Pack 3 in 1 (7877 7929 7913)St...     3451.0   
47  Star Wars Super Pack 3 in 1 (9490, 9492, 9496)...     6769.0   
52  Republic Troopers vs. Sith TroopersClone Troop...     6159.0   
59  Coruscant Police GunshipRepublic AV-7 Anti-Veh...     8293.0   
67  Rey's Speeder™SnowspeederMini Slave 1Imperial ...    11410.0   
76  Krennic’s Imperial ShuttleCaptain Rex’s AT-TED...     6934.0   
82  Wonder Woman" Warrior BattleBatmanThe MilanoTh...    13123.0   

                                           theme_name     id  is_licensed  \
0   Star Wars Episode 1Star Wars Episode 1Star War...   2054           13   
2   Star Wars Episode 4/5/6Star Wars Episode 4/5/6...   4108           26   
4   Star Wars Episode 4/5/6Star Wars Episode 1Star...   2212           14   
6   Star WarsStar Wars Episode 2Star Wars Episode ...   4424           28   
9   MiniMiniStar Wars Episode 2Star Wars OtherStar...   5056           32   
12  Star Wars Episode 4/5/6Star Wars Episode 4/5/6...   3160           20   
16  Star Wars Episode 3Star Wars Episode 3Star War...   4424           28   
20  Star Wars Episode 4/5/6Star Wars Episode 4/5/6...   1738           11   
24  Star Wars Episode 1Star Wars Episode 4/5/6Star...   2528           16   
28  Star Wars Clone WarsStar Wars OtherStar Wars O...   3634           23   
32  Star Wars Clone WarsStar Wars Episode 3Star Wa...   6162           39   
36  Star Wars Episode 4/5/6Star Wars Episode 4/5/6...   4740           30   
42  Star Wars Episode 1Star WarsStar Wars Clone Wa...   5056           32   
47  Star WarsStar WarsSt

In [ ]:
new_era = 2017

## Second answer
2)The year in which Star Wars was not the most popular licensed theme (in terms of numbers of sets released that year) was 2017.

# Solving the third task:


In [ ]:
clean_df = merged[~merged['set_num'].isnull()].copy()


clean_df['count'] = 1
sets_per_year= clean_df.groupby(['year']).sum().reset_index()[['year','count']]
sets_per_year

year  count
0   1950      7
1   1953      4
2   1954     14
3   1955     28
4   1956     12
..   ...    ...
61  2013    593
62  2014    715
63  2015    670
64  2016    608
65  2017    438

[66 rows x 2 columns]

## Third answer

In [ ]:
for index,row in sets_per_year.iterrows():
  print(row['year'],row['count'])

1950 7
1953 4
1954 14
1955 28
1956 12
1957 21
1958 42
1959 4
1960 3
1961 17
1962 40
1963 18
1964 11
1965 10
1966 89
1967 21
1968 25
1969 69
1970 29
1971 45
1972 38
1973 68
1974 39
1975 31
1976 68
1977 92
1978 73
1979 82
1980 88
1981 79
1982 76
1983 57
1984 76
1985 139
1986 123
1987 209
1988 68
1989 114
1990 85
1991 106
1992 115
1993 111
1994 128
1995 128
1996 144
1997 194
1998 325
1999 300
2000 327
2001 339
2002 447
2003 415
2004 371
2005 330
2006 283
2007 319
2008 349
2009 403
2010 444
2011 502
2012 615
2013 593
2014 715
2015 670
2016 608
2017 438
